diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b8d2abc..8b2790b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -373,6 +373,47 @@ FIND_PACKAGE(MMG)
 IF(MMG_FOUND)
 
   SET(HAVE_MMG TRUE CACHE BOOL "Has Mmg tools for Elmer/Ice Mesh adaptation" )
+  ## https://stackoverflow.com/questions/32183975/how-to-print-all-the-properties-of-a-target-in-cmake/56738858#56738858
+  ## https://stackoverflow.com/a/56738858/3743145
+
+  ## Get all properties that cmake supports
+  execute_process(COMMAND cmake --help-property-list OUTPUT_VARIABLE CMAKE_PROPERTY_LIST)
+  ## Convert command output into a CMake list
+  STRING(REGEX REPLACE ";" "\\\\;" CMAKE_PROPERTY_LIST "${CMAKE_PROPERTY_LIST}")
+  STRING(REGEX REPLACE "\n" ";" CMAKE_PROPERTY_LIST "${CMAKE_PROPERTY_LIST}")
+
+  list(REMOVE_DUPLICATES CMAKE_PROPERTY_LIST)
+
+  function(print_target_properties tgt)
+      if(NOT TARGET ${tgt})
+            message("There is no target named '${tgt}'")
+          return()
+      endif()
+
+          foreach (prop ${CMAKE_PROPERTY_LIST})
+          string(REPLACE "<CONFIG>" "${CMAKE_BUILD_TYPE}" prop ${prop})
+          get_target_property(propval ${tgt} ${prop})
+          if (propval)
+              message ("${tgt} ${prop} = ${propval}")
+              endif()
+          endforeach()
+          get_target_property(links ${tgt} INTERFACE_LINK_LIBRARIES)
+          if (links)
+              foreach(link ${links})
+		  list(FIND visited_links ${link} _index)
+		  if(${_index} EQUAL -1)
+                      print_target_properties(${link})
+		      list(APPEND visited_links ${link})
+		  endif()
+              endforeach()
+          endif()
+  endfunction(print_target_properties)
+
+  print_target_properties(Mmg::libmmg3d_so)
+  print_target_properties(Mmg::libmmg3d_so)
+
+  get_target_property(MMG_LIBRARY Mmg::libmmg3d_so LOCATION)
+  get_target_property(MMG_INCLUDE_DIR Mmg::libmmg3d_so INTERFACE_INCLUDE_DIRECTORIES)
   GET_FILENAME_COMPONENT(MMG_LIBDIR ${MMG_LIBRARY} DIRECTORY)
 
   MARK_AS_ADVANCED(HAVE_MMG)
